_drafts/herencia-y-composición.md

```diff
@@ -0,0 +1,324 @@
+---
+layout: post
+title: Herencia y Composición
+author: juliogarciag
+tags:
+    - oop
+    - ruby
+---
+Antes de embarcarnos en explicar qué es la composición, hay que explicar el por qué de este post. La idea es entender *qué* es la composición y por qué existe la frase (extraída del libro [Design Patterns](https://en.wikipedia.org/wiki/Design_Patterns)):
+
+> Favor 'object composition' over 'class inheritance'.
+
+Pensemos primero en un problema: Tenemos una función de un chat que recibe un mensaje (quien sabe cómo aunque puede ser por websockets o por long polling) y debe notificarlo inmediatamente a varios lugares:
+
+```ruby
+class Notifier
+  def initialize(raw_message)
+    self.raw_message = raw_message
+  end
+
+  def send_notifications
+    send_by_websockets(create_event_for_websockets)
+    notify_third_party_service(raw_message)
+  end
+
+  private
+
+  attr_accessor :raw_message
+
+  def send_by_websockets
+    # ...
+  end
+
+  def create_event_for_websockets
+    # ...
+  end
+
+  # ... more private methods
+end
+```
+
+Unas semanas después, tenemos que implementar una feature super genial que lee frases tipo: i want 1 cup of coffee que sean enviadas a ciertos usuarios especiales (bots) y generen órdenes de compra de varios bienes y servicios. Sabemos que tenemos que parsear esa data y sabemos que esto puede resultar complicado, así que creamos un mixin:
```

Comment:
podrias poner la frase "i want 1 cup of coffee" entre comilla, cursiva o algo asi.... para destacarla